reference/src/representation/c_abi.md

```diff
@@ -0,0 +1,74 @@
+# Layout requirements of the C ABI
+
+**Disclaimer:** This chapter is a minimal summary of issue [#44] about Rust's
+current layout requirements of the platforms' C ABI.
+
+Rust programs can target platforms that do not have a C implementation. However,
+when targeting these platforms, attempting to interface with C, e.g., via
+`extern "C"` is illegal, since there is no C ABI to interface with. 
+
+Interfacing with C is allowed in platforms that have a C implementation that
+satisfies all requirements specified in this document.
+
+## Requirements on the C implementation 
+
+If the platform has a C implementation that can be interfaced with via C FFI,
+this implementation is required to conform to the [latest version of the C
+standard]. The following are extra requirements that Rust imposes on top of the
```

Comment:
The problem I see here is that, if we do not require conformance with a specific standard version, all bets are off unless we write down everything that we do require, which to me at least seems like a pretty open question. 

It might be easier to word which specific deviations from the standard we support. We can start by requiring full conformance, and relax this in a backwards compatible way later as we see fit.

Reply:
Considering we have https://gankro.github.io/blah/rust-layouts-and-abis/, aren't we most of the way to describing everything we need?

Keep in mind that a lot of what we're dependent on is described in the ABI documentation itself, in particular the calling convention, limitations on how the stack may be used (if there is one), etc.

Reply:
@briansmith I've pushed a commit adding two unresolved questions about targets without conforming C implementations. We should definitely discuss this in more detail, and resolve what do we exactly need to require here.